src/main/java/nus/climods/logic/parser/parameters/optional/OptionalArgument.java

```diff
@@ -0,0 +1,28 @@
+package nus.climods.logic.parser.parameters.optional;
+
+import org.apache.commons.cli.Option;
+
+/**
+ * Class representing one OptionalArgument e.g --user
+ * Currently uses commons.cli.Option internally to represent an option
+ */
+public class OptionalArgument {
```

Comment:
Actually can we just have each command parser maintain it's own `Options` instead of having a extra wrapper on top of it? Because right now this `OptionalArgument` seems to just building a single `Option` when we can directly build the entire command structure with `Options` directly for each parser.

Something like:

```java
Options options = new Options();

// add t option
options.addOption("t", false, "display current time");
```

> From: https://commons.apache.org/proper/commons-cli/usage.html

This way we can isolate bugs easier, instead of having to work our way down the abstractions. What do you think?

Reply:
We can of course still seperate our common flags like `--user` etc.

Reply:
Thanks for pointing out. I had doubts about this too, decided to include it in case we wanted to not rely on the library directly. I can remove it and make the static class instead